### PR TITLE
Adds Flutter 3.3 support for flutter plugins library manager

### PR DESCRIPTION
**Pull request by wagner1343**
Since Flutter 3.3 flutter plugins library manager doesn't updates flutter plugins library correctly (see #6330).

This pull request fixes that by watching both .packages and .dart_tool/package_config.json files. It also uses dart plugin method to extract path map from package_config.json.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: flutter-idea/src/io/flutter/pub/PubRoot.java [hunk 1]
```diff
@@ -279,6 +281,20 @@ public VirtualFile getPackagesFile() {
     return null;
   }
 
+  public @Nullable Map<String, String> getPackagesMap() {
+    final var packageConfigFile = getPackageConfigFile();
+    if(packageConfigFile != null) {
```
Comment (stevemessick) [inline flutter-idea/src/io/flutter/pub/PubRoot.java:286]: "if (" -- needs a space

File: flutter-idea/src/io/flutter/pub/PubRoot.java [hunk 2]
```diff
@@ -279,6 +281,20 @@ public VirtualFile getPackagesFile() {
...
+      return DotPackagesFileUtil.getPackagesMapFromPackageConfigJsonFile(packageConfigFile);
+    }
+
+    final var packagesFile = getPackagesFile();
+    if(packagesFile != null) {
```
Comment (stevemessick) [inline flutter-idea/src/io/flutter/pub/PubRoot.java:291]: "if ("

File: flutter-idea/src/io/flutter/sdk/FlutterPluginsLibraryManager.java [hunk 3]
```diff
@@ -85,16 +85,23 @@ protected PersistentLibraryKind<FlutterPluginLibraryProperties> getLibraryKind()
...
+    return file.getName().equals(DotPackagesFileUtil.PACKAGE_CONFIG_JSON)
+           && parent != null
+           && parent.getName().equals(DotPackagesFileUtil.DART_TOOL_DIR);
+  }
 
-    if (pubspec != null) {
-      scheduleUpdate();
-    }
+  private void fileChanged(@NotNull final Project project, @NotNull final VirtualFile file) {
+    if(!isPackageConfigFile(file) && !isPackagesFile(file)) return;
```
Comment (stevemessick) [inline flutter-idea/src/io/flutter/sdk/FlutterPluginsLibraryManager.java:101]: "if ("

File: flutter-idea/src/io/flutter/sdk/FlutterPluginsLibraryManager.java [hunk 4]
```diff
@@ -132,24 +139,12 @@ private static Set<String> getFlutterPluginPaths(List<PubRoot> roots) {
...
-        // This is just a backup in case we need it. It does not have a proper cache, but the Dart plugin does.
-        map = loadPackagesMap(configFile);
-      }
-      else {
-        map = DotPackagesFileUtil.getPackagesMap(pubRoot.getPackagesFile());
-        if (map == null) {
-          continue;
-        }
+      final var packagesMap = pubRoot.getPackagesMap();
+      if(packagesMap == null) {
```
Comment (stevemessick) [inline flutter-idea/src/io/flutter/sdk/FlutterPluginsLibraryManager.java:143]: "if ("

**Review by stevemessick — CHANGES_REQUESTED** [review]
Thanks for the contribution! There are a few minor formatting errors, but otherwise this looks good. I'll patch it in and test it once the formatting is done.

**Comment by wagner1343** [comment]
Fixed those formatting errors, thanks for the review!

**Review by stevemessick — APPROVED** [review]
Thanks again!

**Pull request merged**